### PR TITLE
[Do Not Merge] Draft, Submission and Form listing components

### PR DESCRIPTION
**Pull request by virresh**
UI Snapshots:

Drafts and Submissions:
image

Search and Lister
image

PR for review only

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

**Comment by virresh** [comment]
Pending and open Items:
- Documentation of customisation
- Usage of SlingHttpRequest as argument in operation interface
- Using dam path or form path for render
- Behaviour for forms which have been deleted / cancelled agreements

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/formsportal/DiscardDraftOperation.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        return null;
+    }
+
+    @Override
+    public DraftsAndSubmissions.TypeEnum getType() {
+        return DraftsAndSubmissions.TypeEnum.DRAFT;
+    }
+
+    @Override
+    public OperationResult execute(String modelID) {
```
Comment (sakshi-arora1) [inline bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/formsportal/DiscardDraftOperation.java:65]: request parametr map

Comment (virresh) [reply bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/formsportal/DiscardDraftOperation.java]: Changed to RequestParameter Map.
Please re-review

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/PassThroughServlet.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import org.apache.http.impl.client.HttpClients;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.servlets.SlingAllMethodsServlet;
+import org.osgi.service.component.annotations.Component;
+
+@Component(
+    service = { Servlet.class },
+    property = {
+        "sling.servlet.paths=/bin/ptp/avoidcors",
```
Comment (rismehta) [inline bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/PassThroughServlet.java:39]: Why is this required, no path based servlets and this is not as per the HTTP API spec too.

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/models/formsportal/DraftsAndSubmissions.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+public interface DraftsAndSubmissions extends PortalLister {
+
+    /**
+     * Returns type - DRAFT / SUBMISSION / PENDING_SIGN
+     *
+     * @return Type - DRAFT / SUBMISSION / PENDING_SIGN
+     * @since com.adobe.cq.forms.core.components.models.formsportal 2.1.0
+     */
+    default String getType() {
```
Comment (rismehta) [inline bundles/core/src/main/java/com/adobe/cq/forms/core/components/models/formsportal/DraftsAndSubmissions.java:28]: This needs a discussion, why is draftsAndSubmissions composed of operations ?

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/models/formsportal/PortalLister.java [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Return search results and data required for pagination.
+     *
+     * @return Map of key to object
+     * @since com.adobe.cq.forms.core.components.models.formsportal 2.1.0
+     */
+    default Map<String, Object> getSearchResults() {
```
Comment (rismehta) [inline bundles/core/src/main/java/com/adobe/cq/forms/core/components/models/formsportal/PortalLister.java:70]: On first render, there would be no search results, why is it named as search results ? Also, should a lister component have search result as part of its model ?

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/PassThroughServlet.java (hunk 2 above)
Comment (virresh) [reply bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/PassThroughServlet.java:39]: My Bad. This servlet was not supposed to exist here. Was only experimental for testing file attachment preview title. Will remove it right away.

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/models/formsportal/Operation.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+public interface Operation {
+    public static final String OPERATION_MODEL_ID = "operation_model_id";
+
+    public String getName();
+
+    public String getTitle();
+
+    public DraftsAndSubmissions.TypeEnum getType();
+
+    public OperationResult execute(RequestParameterMap parameterMap);
```
[the remark this answers is not in the document]
Comment (virresh) [reply bundles/core/src/main/java/com/adobe/cq/forms/core/components/models/formsportal/Operation.java]: Changed to Map, however it introduces explicit type casting because Request Parameter Map implements Map<String, RequestParameter[]> where RequestParameter is a library introduced POJO.

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/formsportal/DraftsAndSubmissionsImpl.java [hunk 6]
```diff
@@ -0,0 +1,348 @@
...
+        return item;
+    }
+
+    private Boolean isOperationCall() {
+        String operationName = getOperationName();
+        return StringUtils.isNotEmpty(operationName);
+    }
+
+    @Override
+    public Operation.OperationResult getOperationResult() {
```
[the remark this answers is not in the document]
Comment (virresh) [reply bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/formsportal/DraftsAndSubmissionsImpl.java:163]: This is because sling model is re-used for executing operations.
Other way discussed would be to use a different servlet. Currently we're going forward with re-using model as discussed.

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/models/formsportal/DraftsAndSubmissions.java (hunk 3 above)
Comment (virresh) [reply bundles/core/src/main/java/com/adobe/cq/forms/core/components/models/formsportal/DraftsAndSubmissions.java:30]: https://github.com/adobe/aem-core-forms-components/pull/40#discussion_r702673659

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/models/formsportal/PortalLister.java (hunk 4 above)
Comment (virresh) [reply bundles/core/src/main/java/com/adobe/cq/forms/core/components/models/formsportal/PortalLister.java]: Changed to getElements

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/formsportal/DiscardDraftOperation.java [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+import com.adobe.cq.forms.core.components.models.formsportal.DraftsAndSubmissions;
+import com.adobe.cq.forms.core.components.models.formsportal.Operation;
+import com.adobe.fd.fp.api.exception.FormsPortalException;
+import com.adobe.fd.fp.api.service.DraftService;
+
+@Component(
+    service = Operation.class,
+    immediate = true)
+public class DiscardDraftOperation implements Operation {
```
[the remark this answers is not in the document]
Comment (virresh) [reply bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/formsportal/DiscardDraftOperation.java]: Moved to `internal.services.formsportal` subpackage

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/formsportal/DraftsAndSubmissionsImpl.java [hunk 8]
```diff
@@ -0,0 +1,354 @@
...
+        if (!isOperationCall()) {
+            return null;
+        }
+        Operation operation = operationManager.getOperation(getOperationName());
+        if (operation == null) {
+            return null;
+        }
+        return operation.execute(request.getRequestParameterMap());
```
Comment (sakshi-arora1) [inline bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/formsportal/DraftsAndSubmissionsImpl.java:171]: we are still passing requestParameterMap

Comment (virresh) [reply bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/formsportal/DraftsAndSubmissionsImpl.java:171]: requestParameterMap is implementing the Map interface, so I'm passing it directly
It's modified in the interface

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/services/formsportal/DiscardDraftOperation.java [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+        return null;
+    }
+
+    @Override
+    public DraftsAndSubmissions.TypeEnum getType() {
+        return DraftsAndSubmissions.TypeEnum.DRAFT;
+    }
+
+    @Override
+    public OperationResult execute(Map<String, String[]> parameterMap) {
```
Comment (sakshi-arora1) [inline bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/services/formsportal/DiscardDraftOperation.java:73]: String, Object

Comment (virresh) [reply bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/services/formsportal/DiscardDraftOperation.java:73]: https://docs.oracle.com/javaee/6/api/javax/servlet/ServletRequest.html
https://sling.apache.org/apidocs/sling7/org/apache/sling/api/SlingHttpServletRequest.html#getRequestParameterMap--

None of these return <String, Object>

Comment (virresh) [reply bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/services/formsportal/DiscardDraftOperation.java:73]: Discussed seperately. This will be kept as Map<String, String[]>

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/Utils.java [hunk 10]
```diff
@@ -14,11 +14,16 @@
 
 package com.adobe.cq.forms.core.components.internal.models.v1;
 
+import java.net.URISyntaxException;
```
Comment (rismehta) [inline bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/Utils.java:17]: Can we move this to a different package ?

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/Utils.java [hunk 11]
```diff
@@ -42,4 +47,17 @@ private Utils() {}
     public static String generateId(String prefix, String path) {
         return StringUtils.join(prefix, ID_SEPARATOR, StringUtils.substring(DigestUtils.sha256Hex(path), 0, 10));
     }
-}
\ No newline at end of file
+
+    public static String generateActionURL(String modelID, String opName, String requestURI) {
```
Comment (rismehta) [inline bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/Utils.java:51]: This looks very specific utility, its not a generic utility which can be used by all core components. Can we fix this by moving this

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/formsportal/DraftsAndSubmissionsImpl.java [hunk 12]
```diff
@@ -0,0 +1,358 @@
...
+    public String getTitle() {
+        if (isOperationCall()) {
+            return null;
+        }
+        return super.getTitle();
+    }
+
+    @Override
+    public String getLayout() {
+        if (isOperationCall()) {
```
Comment (rismehta) [inline bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/formsportal/DraftsAndSubmissionsImpl.java:251]: This does not look good to me. Looks like a side effect, function is trying to do something which it is not intended to do. Can we fix this

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/Utils.java (hunk 10 above)
Comment (rismehta) [reply bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/Utils.java:17]: Move Utils under internal, it should not be part of model

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/models/services/formsportal/OperationManager.java [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+ */
+
+package com.adobe.cq.forms.core.components.models.services.formsportal;
+
+import java.util.List;
+
+import com.adobe.cq.forms.core.components.models.formsportal.DraftsAndSubmissions;
+import com.adobe.cq.forms.core.components.models.formsportal.PortalLister;
+
+public interface OperationManager {
```
Comment (rismehta) [inline bundles/core/src/main/java/com/adobe/cq/forms/core/components/models/services/formsportal/OperationManager.java:24]: I don't think anyone can customize this interface, please add the related annotation here

File: bundles/core/src/test/resources/searchlister/searchlister-v1-withResults.json [hunk 14]
```diff
@@ -0,0 +1,21 @@
+{
+    "id": "searchlister-fb0b07e803",
+    "title": "Sample Title",
+    "layout": "Custom",
+    "elements": {
+        "pagination": {
+            "loadAction": "/?searchText=Search+Text&limit=10&description=Desc&orderby=title&sort=asc&title=Sample+Title&tags=full%2Fpath%2Fto%2Ftag&offset=1",
```
Comment (rismehta) [inline bundles/core/src/test/resources/searchlister/searchlister-v1-withResults.json:7]: Why is this called as loadAction ? What does this mean

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/commons/v1/clientlibs/layouts/iteminjector/js/itemapi.js [hunk 15]
```diff
@@ -0,0 +1,272 @@
...
+        var link = event.currentTarget.getAttribute('link');
+        window.open(link, "_blank");
+    },
+    operationClickHandler = function (event) {
+        var operation_dataStr = event.currentTarget.getAttribute('operation_data'),
+            operation_data = JSON.parse(operation_dataStr);
+        operationHandlers[operation_data.name].call(this, event, operation_data);
+    };
+
+    class Menu {
```
Comment (rismehta) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/commons/v1/clientlibs/layouts/iteminjector/js/itemapi.js:92]: This won't work on IE.

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/commons/v1/clientlibs/layouts/iteminjector/js/itemapi.js [hunk 16]
```diff
@@ -0,0 +1,272 @@
...
+        componentApp.initializeItemAPI(ItemAPI);
+    };
+
+    $(window).on("core-forms-itemapi-onload", onScriptLoad);
+
+    var registerOperation = function (event, operationConfig) {
+        operationHandlers[operationConfig.name] = operationConfig.handler;
+    }
+
+    $(window).on("core-forms-register-operation", registerOperation);
```
Comment (rismehta) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/commons/v1/clientlibs/layouts/iteminjector/js/itemapi.js:271]: You would have to document this event in the component documentation

**Review by rismehta — CHANGES_REQUESTED** [review]

File: bundles/core/src/test/resources/searchlister/searchlister-v1-withResults.json (hunk 14 above)
Comment (virresh) [reply bundles/core/src/test/resources/searchlister/searchlister-v1-withResults.json:7]: This is for loading next page of results 
(load more button in the UI)

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/commons/v1/clientlibs/layouts/iteminjector/js/itemapi.js (hunk 15 above)
Comment (virresh) [reply ui.apps/src/main/content/jcr_root/apps/core/fd/components/commons/v1/clientlibs/layouts/iteminjector/js/itemapi.js:92]: @sakshi-arora1 any reason for keeping it as Class or should we move this to a js object ?

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/formsportal/DraftsAndSubmissionsImpl.java (hunk 12 above)
Comment (virresh) [reply bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/formsportal/DraftsAndSubmissionsImpl.java:251]: Only other way I can think is to create a different servlet (with a different selector) for performing actions.

cc @sakshi-arora1

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/Utils.java (hunk 11 above)
Comment (virresh) [reply bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/Utils.java:51]: Will make this private to DraftsAndSubmissions Implementation.

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/commons/v1/itemTemplates/card.html [hunk 17]
```diff
@@ -0,0 +1,20 @@
+<template data-sly-template.layout="${@ hideTime, hideFooter}">
+    <div data-cmp-hook-item-template="includes">
+        <sly data-sly-use.clientlib="${'/libs/granite/sightly/templates/clientlib.html'}"
+             data-sly-call="${clientlib.all @ categories=['core.forms.components.commons.v1.clientlibs.layouts.card']}"></sly>
+    </div>
+    <div class="cmp-portallister-results" data-cmp-hook-item-template="container"></div>
```
Comment (rismehta) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/commons/v1/itemTemplates/card.html:6]: Shouldn't this be cmp-portallister__results as per BEM if the block is cmp-portallister and element is the results ?

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/commons/v1/itemTemplates/list.html [hunk 18]
```diff
@@ -0,0 +1,32 @@
+<template data-sly-template.layout="${@ hideTime}">
+    <div data-cmp-hook-item-template="includes">
+        <sly data-sly-use.clientlib="${'/libs/granite/sightly/templates/clientlib.html'}"
+             data-sly-call="${clientlib.all @ categories=['core.forms.components.commons.v1.clientlibs.layouts.list']}"></sly>
+    </div>
+    <table class="cmp-portallister-results cmp-portallister-results--list" data-cmp-hook-item-template="container">
```
Comment (rismehta) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/commons/v1/itemTemplates/list.html:6]: same as above

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/commons/v1/menuTemplate/menu.html [hunk 19]
```diff
@@ -0,0 +1,18 @@
...
+        <ul class="cmp-portallister__menu--list" data-cmp-hook-menu-template="list" role="menu">
+        </ul>
+    </div>
+    <script data-cmp-hook-menu-template="menuItemTemplate"
+            type="x-template" data-sly-call="${menuItem}">
+    </script>
+</template>
+<template data-sly-template.menuItem>
+    <li class="cmp-portallister__menu--item" data-cmp-hook-menu-template="item" role="menuitem" tabindex="0">
+        <span class="cmp-portallister__menu--itemLabel" data-cmp-hook-menu-template="itemLabel" ></span>
```
Comment (rismehta) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/commons/v1/menuTemplate/menu.html:16]: This should be `cmp-portallister__menu-item-label`

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/commons/v1/menuTemplate/menu.html [hunk 20]
```diff
@@ -0,0 +1,18 @@
...
+    <div class="cmp-portallister__menu" data-cmp-hook-menu-template="container">
+        <ul class="cmp-portallister__menu--list" data-cmp-hook-menu-template="list" role="menu">
+        </ul>
+    </div>
+    <script data-cmp-hook-menu-template="menuItemTemplate"
+            type="x-template" data-sly-call="${menuItem}">
+    </script>
+</template>
+<template data-sly-template.menuItem>
+    <li class="cmp-portallister__menu--item" data-cmp-hook-menu-template="item" role="menuitem" tabindex="0">
```
Comment (rismehta) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/commons/v1/menuTemplate/menu.html:15]: This should be `cmp-portallister__menu-item` ... item is not a modifier here but rather than an element as per BEM

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/formsportal/draftsandsubmissions/v1/draftsandsubmissions/draftsandsubmissions.html [hunk 21]
```diff
@@ -0,0 +1,37 @@
...
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/-->
+
+<sly data-sly-use.model="com.adobe.cq.forms.core.components.models.formsportal.DraftsAndSubmissions"
+     data-sly-use.logic="${'draftsandsubmissions.js' @ layout=model.layout}"
+     data-sly-use.clientlib="${'/libs/granite/sightly/templates/clientlib.html'}"
+     data-sly-use.items="${logic.layoutPath}"
+     data-sly-use.menu="${logic.menuPath}"></sly>
+<div class="cmp-draftsandsubmissions" data-cmp-is="draftsandsubmissions" id="${model.id}" data-queryPath="${resource.path}.model.json">
```
Comment (rismehta) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/formsportal/draftsandsubmissions/v1/draftsandsubmissions/draftsandsubmissions.html:22]: Even this is incorrect, can you please fix the BEM selectors by analyzing all of them ?

**Review by rismehta — CHANGES_REQUESTED** [review]

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/Utils.java (hunk 11 above)
Comment (virresh) [reply bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/Utils.java:51]: Moved to another utility class under `internal.services.formsportal`

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/commons/v1/itemTemplates/card.html (hunk 17 above)
Comment (virresh) [reply ui.apps/src/main/content/jcr_root/apps/core/fd/components/commons/v1/itemTemplates/card.html:6]: I intended the whole `cmp-portallister-results` as a seperate block and `cmp-portallister` as a seperate block. The first one is container (it'll be a table in list view) which holds all the listings.
Second one is a single listing and has it's seperate elements. 

Let's discuss this if the rational isn't clear.

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/formsportal/draftsandsubmissions/v1/draftsandsubmissions/draftsandsubmissions.html (hunk 21 above)
Comment (virresh) [reply ui.apps/src/main/content/jcr_root/apps/core/fd/components/formsportal/draftsandsubmissions/v1/draftsandsubmissions/draftsandsubmissions.html]: Updated class names
Please review

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/commons/v1/clientlibs/layouts/iteminjector/js/itemapi.js (hunk 15 above)
Comment (virresh) [reply ui.apps/src/main/content/jcr_root/apps/core/fd/components/commons/v1/clientlibs/layouts/iteminjector/js/itemapi.js]: Moved to object notation

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/models/services/formsportal/OperationManager.java (hunk 13 above)
Comment (virresh) [reply bundles/core/src/main/java/com/adobe/cq/forms/core/components/models/services/formsportal/OperationManager.java:27]: Added provider type

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/formsportal/DraftsAndSubmissionsImpl.java (hunk 12 above)
Comment (virresh) [reply bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/formsportal/DraftsAndSubmissionsImpl.java]: Created servlet

**Review by sakshi-arora1 — COMMENTED** [review]
dispatcher config is missing

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/commons/v1/itemTemplates/card.html (hunk 17 above)
Comment (virresh) [reply ui.apps/src/main/content/jcr_root/apps/core/fd/components/commons/v1/itemTemplates/card.html]: Updated to follow one block per component convention

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/commons/v1/clientlibs/layouts/iteminjector/js/itemapi.js (hunk 16 above)
Comment (virresh) [reply ui.apps/src/main/content/jcr_root/apps/core/fd/components/commons/v1/clientlibs/layouts/iteminjector/js/itemapi.js:277]: Added

**Comment by virresh** [comment]
Please re-review. Updated BEM naming to follow one block per component rule
Dispatcher configuration to be added as part of product.

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/commons/v1/clientlibs/layouts/list/css/list.less [hunk 22]
```diff
@@ -0,0 +1,93 @@
+.cmp-portallister__item--list {
```
Comment (rismehta) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/commons/v1/clientlibs/layouts/list/css/list.less:1]: How can list be a modifier ? Shouldn't this be cmp-portallister__item-list ? Can you please check all such usages and fix them before creating a new PR

**Review by rismehta — CHANGES_REQUESTED** [review]
-1 for now
